mutt/buffer.c

```diff
@@ -388,6 +416,25 @@ size_t buf_strcpy_n(struct Buffer *buf, const char *s, size_t len)
   return buf_addstr_n(buf, s, len);
 }
 
+/**
+ * buf_dequote_comment - Un-escape characters in an email address comment
+ * @param buf Buffer to be un-escaped
+ *
+ * @note the buffer is modified
+ */
+void buf_dequote_comment(struct Buffer *buf)
+{
+  if (!buf)
+    return;
+
+  char *tmp = buf_strdup(buf);
+
+  mutt_str_dequote_comment(tmp);
+
+  buf_strcpy(buf, tmp);
+  FREE(&tmp);
```

Comment:
I don't think we need to dupe the string, here.
`mutt_str_dequote_comment()` only ever shortens the string, so we should be safe.
After calling it, we should call `buf_fix_dptr()` to adjust the end marker.